From ec5ed0f1afe00498f6575c7186d1d628c51d82dc Mon Sep 17 00:00:00 2001
From: Tian Yang <tiany@codeaurora.org>
Date: Mon, 19 Jul 2021 17:14:41 -0700
Subject: [PATCH] map-e: correct draft03 location

In Openwrt 19.07, draft03 is inside json object "data", instead of on the same level.
So this patch aligns with the above assumption.

Signed-off-by: Tian Yang <tiany@codeaurora.org>
Change-Id: Ic2dac4586c58b7cfc67109d1c0a4440b4b95df47
---
 package/network/ipv6/map/files/map.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/package/network/ipv6/map/files/map.sh b/package/network/ipv6/map/files/map.sh
index dcd336be2cde..f27df8c3bb19 100755
--- a/package/network/ipv6/map/files/map.sh
+++ b/package/network/ipv6/map/files/map.sh
@@ -82,7 +82,6 @@ proto_map_setup() {
 		json_add_string mode ipip6
 		json_add_int mtu "${mtu:-1280}"
 		json_add_int ttl "${ttl:-64}"
-		json_add_int draft03 "${draft03:0}"
 		if [ "$mode" = br ]; then
 			json_add_string remote $(eval "echo \$RULE_${k}_IPV6ADDR")
 			json_add_string local $(eval "echo \$RULE_${k}_BR")
@@ -94,6 +93,8 @@ proto_map_setup() {
 		json_add_string link $(eval "echo \$RULE_${k}_PD6IFACE")
 		json_add_object "data"
 			[ -n "$encaplimit" ] && json_add_string encaplimit "$encaplimit"
+			# In Openwrt 19.07, draft03 field is inside "data", so draft03 should be added here.
+			json_add_int draft03 "${draft03:0}"
 			if [ "$maptype" = "map-e" ]; then
 				json_add_array "fmrs"
 				for i in $(seq $RULE_COUNT); do
-- 
2.34.1

